### PR TITLE
Mobile: fixed text issue on some devices

### LINKED ISSUE #2847: Mobile: Text incomplete in one plus phones
In one plus, oppo and some Samsung devices and maybe more that I am not aware of, there is a strange issue in which text is shown incomplete in mobile devices. The reason for this is the shipped font. I tried on one plus 6, If I choose Roboto font, everything is fine and as soon as I changed the font to one plus slate font, some ending characters disappear. This is not the problem with Joplin but with the react-native itself.

[Reference to the issue in react-native](https://github.com/facebook/react-native/issues/15114)

Some possible solutions are to add Roboto font and set it as default.
There is some weird workaround but they are easy and working like because only the last characters are disappearing, add some extra white spaces in the end.
@laurent22  can you please tell me which approach should I use to fix this bug. I would like to solve this.

Some screenshots to show the issue on my one plus phone:
WhatsApp Image 2020-03-22 at 9 25 46 PM
WhatsApp Image 2020-03-22 at 9 25 46 PM (1)

### PR DESCRIPTION
**Pull request by ishantgupta777**
closes #2847 
I changed the default font of those missing text to Roboto and tested the build apk on my phone and now the text is shown properly and everything is fixed. Please have a look.
@PackElend  please label me.

### CODE REVIEW

**Comment by laurent22** [comment]
Did you try this? https://github.com/facebook/react-native/issues/15114#issuecomment-576313266

**Comment by ishantgupta777** [comment]
> Did you try this? [facebook/react-native#15114 (comment)](https://github.com/facebook/react-native/issues/15114#issuecomment-576313266)

Sir,  I think that shouldn't be the proper way to handle this case.  textBreakStrategy takes the overflowing text and puts it in the next line, so that there is no more overflowing. But in our case, I don't think this should be appropriate,  like appearanc coming in one line, and e coming in next line. Even if that worked in one device,  there is no way to test if it works in all devices. But changing in font fanning will make sure, it'll work in all devices.  So should I  change it sir?

**Comment by laurent22** [comment]
My concern is that maybe not all devices have this Roboto font? Do you have some info about this?

> like appearanc coming in one line, and e coming in next line.

Would it really do this with the textBreakStrategy prop? Did you try?

**Comment by ishantgupta777** [comment]
> My concern is that maybe not all devices have this Roboto font? Do you have some info about this?
> 
> > like appearanc coming in one line, and e coming in next line.
> 
> Would it really do this with the textBreakStrategy prop? Did you try?

Yes Sir, I tried and it didn't work on my phone ( one plus6 ), and sir regarding Roboto font not available as far as I know if Roboto font is not available it will fall back to default font available. It will use Roboto only if available.

File: ReactNativeClient/lib/components/screens/config.js [hunk 1]
```diff
@@ -253,6 +254,7 @@ class ConfigScreenComponent extends BaseScreenComponent {
 
 	renderHeader(key, title) {
 		const theme = themeStyle(this.props.theme);
+		theme.headerStyle.fontFamily = 'Roboto';
```
Comment (laurent22) [inline ReactNativeClient/lib/components/screens/config.js]: Please make this change in global-style.js directly.

**Comment by laurent22** [comment]
Ok let's try that then, but I've added a comment to the code.

File: ReactNativeClient/lib/components/screens/config.js (hunk 1 above)
Comment (ishantgupta777) [reply ReactNativeClient/lib/components/screens/config.js]: Done Sir.

**Comment by laurent22** [comment]
Thanks @ishantgupta777

**Comment by laurent22** [comment]
For information, your change unfortunately broke the iOS app so I'm going to revert it.

> regarding Roboto font not available as far as I know if Roboto font is not available it will fall back to default font available

Actually "as far as you know" is not enough. You have to be 100% sure and the way to do that is to test. If you don't have an iOS device to test, you need to find someone who does, and if you cannot find, then we simply should close the issue. We can't take the risk of breaking the app for thousands of users just to fix a minor font glitch.

image

**Comment by ishantgupta777** [comment]
> For information, your change unfortunately broke the iOS app so I'm going to revert it.
> 
> > regarding Roboto font not available as far as I know if Roboto font is not available it will fall back to default font available
> 
> Actually "as far as you know" is not enough. You have to be 100% sure and the way to do that is to test. If you don't have an iOS device to test, you need to find someone who does, and if you cannot find, then we simply should close the issue. We can't take the risk of breaking the app for thousands of users just to fix a minor font glitch.
> 
> image

I am really sorry sir, I totally accept my mistake, I tried to help the community, not to do any harm. I am really sorry for what I did unknowingly. 😢😢 
I will keep these things in mind and won't do such type of mistakes again.

**Comment by laurent22** [comment]
No worries I know you didn't do it on purpose :) Just remember to test on both iOS and Android when changing system-related settings.

**Comment by ishantgupta777** [comment]
> No worries I know you didn't do it on purpose :) Just remember to test on both iOS and Android when changing system-related settings.

I promise sir, I will keep this in mind and also I will install mac os in my windows today itself using Virtualbox and will use the Xbox emulator for IOS testing also.

**Pull request merged**